### PR TITLE
feat: implement product page new design

### PR DESCRIPTION
**Pull request by goldenmaya**
### Jira-link:

https://virtocommerce.atlassian.net/browse/ST-5400
https://virtocommerce.atlassian.net/browse/VCST-105
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.48.0-pr-904-e06a-e06aec27.zip
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.47.0-pr-904-1b70-1b70b730.zip

### CODE REVIEW

File: client-app/shared/catalog/components/product-sidebar.vue [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+const props = defineProps<IProps>();
+
+const { currentCurrency } = useCurrency();
+const { getItemsTotal } = useCart();
+
+const isDigital = computed<boolean>(() => props.product.productType === ProductType.Digital);
+
+const variationsCartTotalAmount = eagerComputed<number>(() => {
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-sidebar.vue:69]: eagerComputed for cheep operations, often boolean
here better to use computed
https://v6-7-6.vueuse.org/shared/eagercomputed/

File: client-app/pages/product.vue [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
+              :product="product"
+              :related-products="relatedProducts"
+              :model="item"
+            />
+          </template>
+        </div>
+
+        <ProductSidebar
+          :class="[
+            'flex-none md:sticky md:top-[4.5rem] md:w-64 lg:top-[6.5rem] xl:w-[17.875rem]',
```
Comment (NaMax66) [inline client-app/pages/product.vue:53]: 17.875 too detailed
0.25 step is ok

File: client-app/shared/catalog/components/product-sidebar.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="space-y-6">
+    <ProductPriceBlock :product="product">
+      <!-- Price & Delivery (with variations) -->
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-sidebar.vue:4]: remove comment or change it (no delivery inside)

File: client-app/shared/catalog/components/product-sidebar.vue [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            {{ currentCurrency.symbol }}{{ variationsCartTotalAmount.toFixed(2) }}
+          </span>
+        </div>
+
+        <VcButton class="mt-4 print:hidden" :to="{ name: 'Cart' }" full-width>
+          {{ $t("pages.product.view_cart_button") }}
+        </VcButton>
+      </template>
+
+      <!-- Price & Delivery (without variations) -->
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-sidebar.vue:22]: remove comment or change it (no delivery inside)

File: client-app/shared/catalog/composables/useProduct.ts [hunk 5]
```diff
@@ -20,6 +24,21 @@ export function useProduct() {
...
+
+    eventItems.forEach(({ productId, inWishlist }) => {
+      if (product.value && product.value.id === productId) {
+        product.value.inWishlist = inWishlist;
+        trigger = true;
+      }
+    });
+
+    if (trigger) {
+      triggerRef(product);
```
Comment (NaMax66) [inline client-app/shared/catalog/composables/useProduct.ts:38]: triggerRef use with shallowRef.
But product uses ref. It is deeply reactive. 
Should work without triggerRef. 
Please try
https://vuejs.org/api/reactivity-advanced.html#triggerref

**Review by NaMax66 — CHANGES_REQUESTED** [review]

File: client-app/pages/product.vue (hunk 2 above)
Comment (goldenmaya) [reply client-app/pages/product.vue:53]: the styles will be moved into the VcLayoutWithSidebar component

**Review by NaMax66 — APPROVED** [review]

**Review by Andrew-Orlov — APPROVED** [review]

File: client-app/ui-kit/components/molecules/button/vc-button.vue [hunk 6]
```diff
@@ -62,7 +62,7 @@ export interface IEmits {
 }
 
 interface IProps {
-  color?: "primary" | "secondary" | "success" | "info" | "neutral" | "warning" | "danger";
+  color?: "primary" | "secondary" | "success" | "info" | "neutral" | "warning" | "danger" | "accent";
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/molecules/button/vc-button.vue:65]: `primary` and `secondary` is already primary and secondary accent colors. What `accept` mean here? Is it tertiary accent?

File: client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue [hunk 7]
```diff
@@ -8,13 +8,14 @@
 import { computed } from "vue";
 
 interface IProps {
-  size?: number;
+  size?: number | string;
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue:11]: As in https://github.com/VirtoCommerce/vc-theme-b2b-vue/pull/912 I propose to leave only string here for clarity

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/shared/catalog/components/product-price-block.vue [hunk 8]
```diff
@@ -3,122 +3,114 @@
...
-          </div>
-
-          <div class="w-1/3">
-            <a
-              :href="mailToLink"
-              target="_blank"
-              class="flex h-14 cursor-pointer select-none items-center justify-center border-r p-2 hover:bg-gray-100"
+      <div class="flex select-none divide-x print:hidden">
+        <AddToList class="flex w-1/5 flex-col justify-stretch" :product="product" tooltip-placement="bottom">
+          <template #default="{ openModal, isInWishlist, isAuthenticated }">
```
Comment (asvishnyakov) [inline client-app/shared/catalog/components/product-price-block.vue:8]: 1. I would suggest to use `isAuthenticated` directly from `useUser` composable.
2. We also have `product` here so we don't need to pass product to modal and then get `product.isInWishlist` back, instead just directly use `product.isInWishlist`.
3. The last one is to open modal. I think we can leave it as is or move to product composable, it's up to you.

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/shared/static-content/components/related-products.vue [hunk 9]
```diff
@@ -1,59 +1,85 @@
...
+        :key="index"
+        :product="item"
+        class="w-[calc((100%-1.75rem)/2)] sm:w-[calc((100%-2*1.75rem)/3)] md:w-[calc((100%-1.75rem)/2)]"
+        @link-click="ga.selectItem(item)"
+      />
     </div>
 
-    <div class="hidden flex-none xl:block xl:w-3/12"></div>
-  </div>
+    <VcCarousel v-else :slides="relatedProducts" :options="relatedProductsCarouselOptions" navigation>
```
Comment (asvishnyakov) [inline client-app/shared/static-content/components/related-products.vue:20]: Let's simplify and use just `carouselOptions`

File: client-app/shared/static-content/components/related-products.vue [hunk 10]
```diff
@@ -1,59 +1,85 @@
...
+    prepend-icon="cube"
+    size="lg"
+    class="order-last print:hidden max-md:-mx-4.5"
+  >
+    <div v-if="lg" class="-mb-2 flex flex-wrap items-stretch gap-x-7 gap-y-4">
+      <ProductCardRelated
+        v-for="(item, index) in mobileProducts"
+        :key="index"
+        :product="item"
+        class="w-[calc((100%-1.75rem)/2)] sm:w-[calc((100%-2*1.75rem)/3)] md:w-[calc((100%-1.75rem)/2)]"
```
Comment (asvishnyakov) [inline client-app/shared/static-content/components/related-products.vue:15]: May be moved it to css and use scss variables to explain what these magic numbers are? (or even use `theme()` is possible)

File: client-app/shared/static-content/components/related-products.vue [hunk 11]
```diff
@@ -1,59 +1,85 @@
...
+        @link-click="ga.selectItem(item)"
```
Comment (asvishnyakov) [inline client-app/shared/static-content/components/related-products.vue:16]: Let's move it to separate handler and call GA inside, then reuse it here and below

File: client-app/ui-kit/components/molecules/button/vc-button.vue (hunk 6 above)
Comment (goldenmaya) [reply client-app/ui-kit/components/molecules/button/vc-button.vue:65]: https://www.figma.com/file/RAvri6to8nVBUDpDQCBvUA/Mercury-UI-kit?type=design&node-id=12936-20638&mode=design&t=01GZ1gehGyAO3ayO-4

File: client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue (hunk 7 above)
Comment (goldenmaya) [reply client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue:11]: You can make another task for that.

File: client-app/shared/static-content/components/related-products.vue (hunk 11 above)
Comment (asvishnyakov) [reply client-app/shared/static-content/components/related-products.vue:16]: https://virtocommerce.atlassian.net/browse/ST-5654

File: client-app/shared/static-content/components/related-products.vue (hunk 10 above)
Comment (asvishnyakov) [reply client-app/shared/static-content/components/related-products.vue:15]: https://virtocommerce.atlassian.net/browse/ST-5654

File: client-app/shared/static-content/components/related-products.vue (hunk 9 above)
Comment (asvishnyakov) [reply client-app/shared/static-content/components/related-products.vue:20]: https://virtocommerce.atlassian.net/browse/ST-5654

File: client-app/shared/catalog/components/product-price-block.vue (hunk 8 above)
Comment (asvishnyakov) [reply client-app/shared/catalog/components/product-price-block.vue:8]: https://virtocommerce.atlassian.net/browse/ST-5654

File: client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue (hunk 7 above)
Comment (asvishnyakov) [reply client-app/ui-kit/components/atoms/hexagon-icon/vc-hexagon-icon.vue:11]: https://virtocommerce.atlassian.net/browse/ST-5653

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**